### ISSUE TITLE
Have `--check` output be in different formats (pretty/unified/json)

### ISSUE THREAD

**Issue by levouh** [JohnnyMorganz/StyLua#230]
In summary, add an argument to dump the formatted file to `STDOUT` rather than doing things in-place, say, `--stdout/-d` mode.

An alternative that I've used with a number of different tools is to pass diff output to `patch` in order to reconstruct what the changed file would look like. For `stylua`, something like `stylua --check /path/to/file.lua | patch -p0 -s -o -`. However, the `--check` option does not output a patch-able format (to my knowledge) so doing something like this doesn't give the "final" output either.

Given the former option of adding a new argument or similar, discerning which `STDOUT` applies to which file can be difficult as well, so it _seems_ absolutely viable to have `stylua` only accept a single file when being run in either of these modes.

Thoughts @JohnnyMorganz?

**Comment by JohnnyMorganz** [comment]
You can pass code into `stdin`, then its formatted output is dumped to `stdout` (`cat /path/to/file.lua | stylua -`). Is this reasonable for your use case, or is it necessary that we take in a path instead?

For `--check` outputing a `patch`-able diff, that might be nice to have. I assume `patch` takes in a git-style diff? If we do this, it also might help editor integrations aswell, where they can patch parts of the file rather than currently replacing the whole file buffer with the formatted output, losing some niceties.

**Comment by levouh** [comment]
>You can pass code into stdin, then its formatted output is dumped to stdout (cat /path/to/file.lua | stylua -). Is this reasonable for your use case,

Awesome, this satisfies 99% of my use case, but it still seems like it'd be worth it to have an option to explicitly pass the output to `STDOUT`.

>or is it necessary that we take in a path instead?

Its effectively the same thing; I know the path so I'm either just outputting the file or passing it to `stylua` directly.

>I assume patch takes in a git-style diff?

Yes, exactly. I'm not familiar with it enough to decoratively say whether or not:

>it also might help editor integrations aswell, where they can patch parts of the file

is something that is easily done, but I cannot imagine it hurts either way. Depends on how easy it is to do and the complexity that it adds to `stylua` itself. If it's easy, etc. then it _might_ be worth it, but if you have the start/end state of the file, it isn't too hard to form a `git`-style diff via some light scripting (implying it is something users can easily add themselves at no cost to the complexity of this project.

---

So TLDR, either of the things you mentioned would be _nice_, but given it is possible to dump a file to `STDOUT` like you mentioned, that is completely fine for my use case (although perhaps you should document it somewhere [unless I just missed it while skimming]).

Feel free to close this if you want, or leave it open to track any changes you see fit.

**Comment by JohnnyMorganz** [comment]
I think I'll leave the formatting dumped to stdout when taking in a path for now, since it can be done externally with `cat` and piping.

Having `--check` output in a more patch-friendly/machine-friendly format seems like a reasonable addition though. Probably through something like `--output pretty/unified/json` where `pretty` is current, `unified` is `patch`-able, `json` is the diff output in json. Will probably add something like that when I have some time.

**Comment by JohnnyMorganz** [comment]
This is now supported in v0.13.0: https://github.com/JohnnyMorganz/StyLua/releases/tag/v0.13.0
